### ISSUE TITLE
Undefined symbols for architecture x86_64

### ISSUE THREAD

**Issue by anudit** [iden3/react-native-rapidsnark#7]
Hey, I'm getting the following error while building an React Native Expo App, all latest versions for iOS.

macos-14-arm64
Xcode 15.3

```
[RUN_FASTLANE] ❌  Undefined symbols for architecture x86_64
┌─ Symbol: _groth16_prover
└─ Referenced from: -[Rapidsnark groth16Prove:witnessData:proofBufferSize:publicBufferSize:errBufferSize:resolve:reject:] in react_native_rapidsnark[x86_64][3](Rapidsnark.o)
[RUN_FASTLANE] 
❌  ld: symbol(s) not found for architecture x86_64
[RUN_FASTLANE] 
❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by 5eeman** [comment]
Hi,
Are you running local build or in Expo CI?

**Comment by anudit** [comment]
Hey @5eeman Both EAS build using Github Actions & Local EAS builds return the same error on `0.0.1-alpha.3`
```
[RUN_FASTLANE] +---------------------------------------+
[RUN_FASTLANE] |           Build environment           |
[RUN_FASTLANE] +-------------+-------------------------+
[RUN_FASTLANE] | xcode_path  | /Applications/Xcode.app |
[RUN_FASTLANE] | gym_version | 2.217.0                 |
[RUN_FASTLANE] | sdk         | iPhoneOS17.4.sdk        |
[RUN_FASTLANE] +-------------+-------------------------+
```

Android builds work perfectly though.

**Comment by demonsh** [comment]
Hi @anudit.
Currently we have  only arm64 build for IOS.

Can you try to build only for arm64 or for device?

**Comment by anudit** [comment]
Hey @demonsh, Expo doesn't offer a way to pick only arm64.
Can I ask what the limitation is for not having x86_64 as it is currently required for the 64-bit iOS simulator.

Thanks!

**Comment by demonsh** [comment]
In radpidsnark we have asm optimisation for mobiles devices. We have builds without optimisations specifically for simulator. As x86_64 required only for old intel devices it was not included in the library. On modern apple devices with arm architecture, already available simulators with arm64, and x86_64 is not reqired for local builds.
<img width="380" alt="image" src="https://github.com/iden3/react-native-rapidsnark/assets/5119683/19683251-04f0-46ad-9d33-ba502fb7ada2">

You can exclude architecture in the settings and it should work
<img width="653" alt="image" src="https://github.com/iden3/react-native-rapidsnark/assets/5119683/a52675cd-8bac-4edb-8b20-94a463e11a84">
<img width="575" alt="image" src="https://github.com/iden3/react-native-rapidsnark/assets/5119683/43d5ecdb-452f-4aea-b36f-33216c7f7e0c">

For devices you will need arm anyway.

If there's a good reason to add x86_64 for simulators we can do it.

**Comment by anudit** [comment]
@demonsh I tried excluding it in the settings and Podfile but still hitting the `Undefined symbols for architecture x86_64` on the simulator build.
```
installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'x86_64'
      end
    end
```
Here is a minimal repro of it [https://github.com/anudit/omnid-repro](https://github.com/anudit/omnid-repro)
You might have to setup an expo account. 

Ik this is an edge case but thanks for help! (other ways as just so much slower than rapidsnark really don't want to switch)

**Comment by demonsh** [comment]
Thanks @anudit. I'll take a look on it.

**Comment by 5eeman** [comment]
@anudit please, try using a version from `fix/ios_sim_x64` branch.
I've updated ios sim libs with version with x86_64 support.

**Comment by anudit** [comment]
Thanks a lot @5eeman  @demonsh iOS Simulator builds work perfectly now on the `fix/ios_sim_x64` branch.
```
[RUN_FASTLANE] › Generating debug Omnid » Omnid.app.dSYM
[RUN_FASTLANE] › Executing Omnid » Bundle React Native code and images
[RUN_FASTLANE] › Executing Omnid » [CP] Copy Pods Resources
[RUN_FASTLANE] › Executing Omnid » [CP] Embed Pods Frameworks
[RUN_FASTLANE] › Signing   Omnid » Omnid.app
[RUN_FASTLANE] › Creating  Omnid » Omnid.app
[RUN_FASTLANE] › Build Succeeded
[UPLOAD_APPLICATION_ARCHIVE] Application archives: /var/folders/lb/p46m2p597tg43_4nfbct5vjc0000gn/T/eas-build-local-nodejs/1fb83023-af03-450b-b689-292ae07e8d4d/build/ios/build/Build/Products/Debug-iphonesimulator/Omnid.app
[UPLOAD_APPLICATION_ARCHIVE] Uploading application archive...
[PREPARE_ARTIFACTS] Preparing artifacts
[PREPARE_ARTIFACTS] Writing artifacts to /Users/anudit/Documents/GitHub/omnidapp/app-ios.tar.gz

Build successful
You can find the build artifacts in /Users/anudit/Documents/GitHub/omnidapp/app-ios.tar.gz
```
![image](https://github.com/iden3/react-native-rapidsnark/assets/12015553/206dbdae-4151-4f90-be4b-0e513ab1dbe2)

**Comment by demonsh** [comment]
@anudit just for my interest, how you are building witness?

**Comment by anudit** [comment]
Webview for now, switching to native c++ soon. Experimenting with https://github.com/0xPolygonID/witnesscalc & https://github.com/philsippl/circom-witness-rs 
Not keen on adding an additional wasm runtime though we can link arkworks directly.

**Comment by demonsh** [comment]
Thanks @anudit 
We are using https://github.com/0xPolygonID/witnesscalc  in production for almost two years.

If you will succeed with https://github.com/philsippl/circom-witness-rs please share your experience. We are working on adding missing instructions to `circom-witness-rs`, but at the moment it is more experiments.

**Comment by demonsh** [comment]
Fixed in [0.0.1-alpha.4](https://github.com/iden3/react-native-rapidsnark/actions/runs/9354535625)
